src/server/checker/CheckerService.ts

```diff
@@ -164,6 +181,97 @@ export class CheckerService {
     return checker
   }
 
+  private findAndCheckAuthOwner: (
+    id: string,
+    user: User | null
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  ) => Promise<(Model<Checker, any> & Checker) | null> = async (id, user) => {
+    const checker = await this.CheckerModel.findByPk(id, {
+      include: [this.UserModel],
+    })
+    if (checker) {
+      const isAuthorized = checker.users?.some(
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
+        (u) => u.UserToChecker.isOwner && u.id === user?.id
+      )
+      if (!isAuthorized) {
+        throw new Error('Unauthorized')
+      }
+    }
+    return checker
+  }
+
+  listCollaborators: (id: string, user: User) => Promise<CollaboratorUser[]> =
+    async (id, user) => {
+      const checker = await this.findAndCheckAuth(id, user)
+      if (checker) {
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
+        return checker.users
+      } else return []
+    }
+
+  addCollaborator: (
+    id: string,
+    user: User,
+    collaboratorEmail: string
+  ) => Promise<void> = async (id, user, collaboratorEmail) => {
+    const transaction = await this.sequelize.transaction()
+    const options = this.isSqliteFile ? {} : { transaction }
+
+    try {
+      const checker = await this.findAndCheckAuth(id, user)
+      collaboratorEmail = collaboratorEmail.toLowerCase()
+
+      if (checker) {
+        const collaboratorUser = await this.UserModel.findOne({
```

Comment:
Same for this, this should be wrapped in a transaction. Otherwise, you might end up with SQL constraint errors when there are concurrent requests trying to add the same user.